### PR TITLE
feat(TF>1.3)!: Min terraform version 1.3. Changed variables type object with optional parameters

### LINKED ISSUE #26: Optional log_config block for nat
It would be nice to have this optional

https://github.com/terraform-google-modules/terraform-google-cloud-router/blob/2154aee587cf302cb583b1334f52f923b1801477/nat.tf#L37-L40

Maybe a simple enhancement like so:

```terraform
  log_config {
    enable = lookup(lookup(each.value, "log_config", {}), "enable", true)
    filter = lookup(lookup(each.value, "log_config", {}), "filter", "ALL")
  }
```

Probably it would be more elegant to be able to drop the whole `log_config{}` with some `dynamic{}` block logic

### LINKED ISSUE #34: type nats property
It's now defined as `any`. What does it take in? Would make sense to type it better to avoid mistakes.

### LINKED ISSUE #50: Properties are not documented well
Several properties on this module are not documented well. An example is `bgp`: an unexplained acronym with an `any` type. I have no information about this property.

Please improve documentation so users don't have to fish through the implementation to determine how to set properties.

### LINKED ISSUE #80: update testing framework
### TL;DR

update testing framework

### Terraform Resources

_No response_

### Detailed design

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by imrannayer**
fixed #50 
fixed #34 
fixed #26 
Fixed #80

### CODE REVIEW

File: examples/nat/main.tf [hunk 1]
```diff
@@ -14,13 +14,23 @@
...
+  project_id   = var.project_id
+  network_name = "test-network"
+  routing_mode = "GLOBAL"
+  subnets      = []
+}
+
+
 # [START cloudnat_simple_create]
 module "cloud_router" {
-  source  = "terraform-google-modules/cloud-router/google"
```
Comment (bharathkkb) [inline examples/nat/main.tf:19]: The above `# [START cloudnat_simple_create]` indicates this is part of a c.g.c doc. Could we keep the registry references? If this is used in a test, we can use module swapper as a build step to swap in local references.

https://github.com/terraform-google-modules/terraform-google-network/blob/6b1334d8b24c12082aa9d9f92e400cc1d16461e3/build/int.cloudbuild.yaml#L17

File: outputs.tf [hunk 2]
```diff
@@ -18,3 +18,8 @@ output "router" {
   value       = google_compute_router.router
   description = "The created router"
 }
+
+output "nat" {
+  value       = google_compute_router_nat.nats
+  description = "The created router"
```
Comment (bharathkkb) [inline outputs.tf:24]:
```suggestion
  description = "Created NATs"
```

File: test/integration/nat/nat_test.go [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+func TestNat(t *testing.T) {
+	nat := tft.NewTFBlueprintTest(t)
+
+	nat.DefineVerify(func(assert *assert.Assertions) {
+		// Getting the expected Project ID from the outputs
+		projectId := nat.GetStringOutput("project_id")
+		// Router Information
+		routerName := nat.GetStringOutput("router_name")
+		routerRegion := nat.GetStringOutput("router_region")
+		fmt.Printf("%s, %s, %s", projectId, routerName, routerRegion)
```
Comment (bharathkkb) [inline test/integration/nat/nat_test.go:35]: nit: If it's debug we can remove? If we want to keep it for some reason, prefer t.Logf()

File: test/integration/simple_example/simple_example_test.go [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+
+	example.DefineVerify(func(assert *assert.Assertions) {
+		// Getting the expected Project ID from the outputs
+		projectId := example.GetStringOutput("project_id")
+		// Router Information
+		routerName := example.GetStringOutput("router_name")
+		routerRegion := example.GetStringOutput("router_region")
+		fmt.Printf("%s, %s, %s", projectId, routerName, routerRegion)
+
+		getRouterName := gcloud.Run(t, fmt.Sprintf("compute routers describe %s  --project %s  --region %s ", routerName, projectId, routerRegion))
```
Comment (bharathkkb) [inline test/integration/simple_example/simple_example_test.go:37]: nit: prefer Runf here and throughout
https://pkg.go.dev/github.com/GoogleCloudPlatform/cloud-foundation-toolkit/infra/blueprint-test@v0.7.0/pkg/gcloud#Runf
```suggestion
		getRouterName := gcloud.Runf(t,"compute routers describe %s  --project %s  --region %s ", routerName, projectId, routerRegion)
```

**Review by bharathkkb — APPROVED** [review]
Thank you so much for updating this! A few minor comments
